inception-pdf-editor/src/main/js/pdfanno/src/pdfanno.js

```diff
@@ -189,6 +211,25 @@ async function displayViewer () {
     textLayer.setup(analyzeResult)
     window.annoPage.pdftxt = analyzeResult
 
+// BEGIN INCEpTION EXTENSION - #802 - Rendering annotations pagewise
+    const renderTimeout = 1500
```

Comment:
@reckart this is the timeout that PDFAnno will wait until it renders a new page if the page hasn't changed since last change.
Do you think 1500 ms is a good value?
I don't know if it is too short or too long.

Reply:
> it renders a new page if the page hasn't changed since last change

I'm not sure I understand this. What does it mean if a page is "new" or has been "changed" and what kind of change is the "last change" - probably a different kind of change than the change to the page...?

Reply:
When you scroll through a PDF document the editor will not render annotations immediately for each page but it will wait for the `renderTimeout` until it renders annotations.
The reason is if you scroll through the document quickly you don't want a render request for each page, but only when you stop at some page and stay on it.
So the question is, how long should you have to wait until editor requests annotations.

Reply:
Things appear to be "interactive" if the reaction time is 200ms or less. In some places in our code, we use a delay of 500ms. Using a longer delay may be sensible. By which rationale did you choose 1500ms?

Reply:
I chose it like this because scrolling through a page takes some time, so if it is too short it might trigger rendering too often.
That was my thought.
So when it renders the page will hang and the user can't continue scrolling through the document.
I think this could be annoying.

Reply:
Is there a way to render n page ahead and start requesting new ones when the user is e.g. half-way through the already rendered pages?

Reply:
Still it would hang though.
Also the rendering process would take longer I guess.
Currently previous and next page is rendered along to the current page so that you can see annotations at page boundaries.

Reply:
I think I found a cooler way.
What about if we render only when page changes and no movement in document is detected for 500ms ?
It feels pretty smooth to me and you don't have to scroll too fast through document if you want to get to a certain point without interruption.
I will commit this, then you can test and tell me then whether you like it or not, ok?

Reply:
It sounds very good. Looking forward to testing it.